### PR TITLE
runmodes: dpdk: Move IPS check outside of the core suricata file and add checksum offload to DPDK v7

### PR DESCRIPTION
**Pull request by lukashino**
This is a follow-up of PR #8330

Changes
- Add LiveDevFinalize prior to IPS check location (needed by the runmode to iterate over LiveDevs)
- Change the location of IPS check by the runmode (due to Segfault when the check was at the very beginning of the function)

Redmine tickets:
https://redmine.openinfosecfoundation.org/issues/5580
https://redmine.openinfosecfoundation.org/issues/5553
https://redmine.openinfosecfoundation.org/issues/5706

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103972 | 81.08% |
| | IPS_AFP_stats_chk | | |
| .tcp.segment_from_pool | 17114563 | 15750518 | 92.03% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 52630 | 0.01 | 

Pipeline 11317

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103905 | 81.03% |
| | IPS_AFP_stats_chk | | |
| .tcp.segment_from_pool | 17114563 | 15752055 | 92.04% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 249 | 0.00 | 

Pipeline 11319

**Comment by lukashino** [comment]
Code in pipeline 11323 changes the way IPS check is called but not the location of where this check is executed (meaning Suricata core files no longer contain runmode-specific code but the check remains at the same place). This also means that the [Redmine ticket 5706](https://redmine.openinfosecfoundation.org/issues/5706) remains unsolved as the check for IPS mode is done after the `http_body_inline` is set. 

However, this PR solves the aforementioned Redmine ticket 5706 and thus enables `cfg_prec->http_body_inline` when IPS mode is active. As a result, I believe there might be a chance that the `.tcp.segment_from_pool` results deviate from the base results but actually in the correct way.

**Comment by victorjulien** [comment]
FreeBSD 13:
```
 CC       runmode-netmap.o
../../../src/runmode-netmap.c:410:5: error: redefinition of 'NetmapRunModeIsIPS'
int NetmapRunModeIsIPS()
    ^
../../../src/runmode-netmap.c:59:12: note: previous definition is here
static int NetmapRunModeIsIPS(void)
           ^
1 error generated.
gmake[3]: *** [Makefile:3477: runmode-netmap.o] Error 1
```

File: src/suricata.c [hunk 1]
```diff
@@ -2663,6 +2639,11 @@ int PostConfLoadedSetup(SCInstance *suri)
 
     MacSetRegisterFlowStorage();
 
+    LiveDeviceFinalize(); // must be after EBPF extension registration
```
Comment (victorjulien) [inline src/suricata.c:2642]: this now runs before `ConfigGetCaptureValue` which relies on the livedevs freed by ` LiveDeviceFinalize`:
```
Starting program: /home/victor/dev/suricata/src/suricata -vvvvv -c /home/victor/dev/qa/ips/tests/geneve//suricata.yaml --af-packet -S /home/victor/dev/qa/ips/tests/geneve//test.rules -l /tmp//geneve-decoder/suricata/
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Notice: suricata: This is Suricata version 7.0.0-rc1-dev (b754887e6 2023-01-17) running in SYSTEM mode [suricata.c:1138]
Info: cpu: CPUs/cores online: 8 [util-cpu.c:182]
Info: af-packet: Setting IPS mode [runmode-af-packet.c:150]

Program received signal SIGSEGV, Segmentation fault.
strlcpy (dst=0x7fffffffe060 " \271\002\021\001\006#", src=0x316634736e65 <error: Cannot access memory at address 0x316634736e65>, siz=128) at util-strlcpyu.c:55
55                  if ((*d++ = *s++) == 0)
(gdb) bt
#0  strlcpy (dst=0x7fffffffe060 " \271\002\021\001\006#", src=0x316634736e65 <error: Cannot access memory at address 0x316634736e65>, siz=128) at util-strlcpyu.c:55
#1  0x0000555555cc510f in ConfigGetCaptureValue (suri=0x5555575b1360 <suricata>) at suricata.c:2435
#2  0x0000555555cc59c9 in PostConfLoadedSetup (suri=0x5555575b1360 <suricata>) at suricata.c:2660
#3  0x0000555555cc65b6 in SuricataMain (argc=9, argv=0x7fffffffe2b8) at suricata.c:2901
#4  0x0000555555cbffdd in main (argc=9, argv=0x7fffffffe2b8) at main.c:22
```

**Comment by lukashino** [comment]
> FreeBSD 13:
> 
> ```
>  CC       runmode-netmap.o
> ../../../src/runmode-netmap.c:410:5: error: redefinition of 'NetmapRunModeIsIPS'
> int NetmapRunModeIsIPS()
>     ^
> ../../../src/runmode-netmap.c:59:12: note: previous definition is here
> static int NetmapRunModeIsIPS(void)
>            ^
> 1 error generated.
> gmake[3]: *** [Makefile:3477: runmode-netmap.o] Error 1
> ```

Dang, forgot to delete one function definition :/ Fixed.

File: src/suricata.c (hunk 1 above)
Comment (lukashino) [reply src/suricata.c:2642]: Ah thanks, didn't catch that without the sanitizer enabled. 
Removed functions that work with pre-registered devices and used functions that work with registered devices only (as we are after `LiveDeviceFinalize` function).

**Comment by lukashino** [comment]
#8394 is a follow-up

**Pull request closed without merging**